test/testImageCommands.sh

```diff
@@ -1,82 +1,89 @@
+# import functions that are shared across unit tests
+source PharoLauncherCommonFunctions.sh
 
-#setup pharo launcher and image name paths
-PHLSCRIPT=./pharo-launcher.sh
-IMAGE=./shared/PharoLauncher.image
+#ensure that Shell unit test library is installed
+ensureShunitIsPresent
 
-# setup commands for image manipulation
-processListCommand="runLauncherScript image processList"
-launchCommand="runLauncherScript image launch PhLTestImage"
-killAllCommand="runLauncherScript image kill --all"
-killCommand="runLauncherScript image kill PhLTestImage"
+#setup sample image name and template name
+SAMPLEIMAGE="PhLTestImage"
+SAMPLETEMPLATE="Pharo 8.0 - 64bit (old stable)"
 
-oneTimeSetUp() {
-	./createSampleImageCmd.sh
-	prepareLauncherScriptAndImage
+# setup commands for sample image manipulation
+createSampleImageCommand () {
+    runLauncherScript image create $SAMPLEIMAGE "`$SAMPLETEMPLATE`"
+}
+
+launchSampleImageCommand () {
+    runLauncherScript image launch $SAMPLEIMAGE
+}
+
+killSampleImageCommand () {
+    runLauncherScript image kill $SAMPLEIMAGE
 }
 
-runLauncherScript() {
-	pushd ..
-	$PHLSCRIPT $@
-	popd
+deleteSampleImageCommand () { 
+    runLauncherScript image delete $SAMPLEIMAGE
 }
 
-prepareLauncherScriptAndImage () {
-	# ensure that launcher script and image is in needed directories for test evaluation (before packaging)
-	pushd ..
-	if [ ! -f "$PHLSCRIPT" ] ; then
-	    cp ./script/pharo-launcher.sh $PHLSCRIPT
-	fi
-	if [ ! -f "$IMAGE" ] ; then
-		mkdir -p shared
-	    cp PharoLauncher.image $IMAGE
-	fi
-	popd
+processListCommand () {
+    runLauncherScript image processList
 }
 
-cleanupLauncherScriptAndImage () {
-	pushd ..
-	rm -rf ./shared
-	rm -f $PHLSCRIPT
-	popd
+killAllCommand () {
+    runLauncherScript image kill --all
+}
+
+
+
+oneTimeSetUp() {
+	prepareLauncherScriptAndImage
+	createSampleImageCommand
 }
 
 testLauncherProcessListCommandWhenNoPharoImageRunningShouldReturnEmptyList(){
-	result=$($processListCommand)
-	assertNull "$result"
+	result=$(processListCommand)
+	#since VM prints some warnings, we need to check presence of image name from process list
+	assertNotContains "$result" "$SAMPLEIMAGE"
 }
 
 testLauncherProcessListCommandWhenImageIsLaunchedShouldReturnOneImage(){
-	$launchCommand >/dev/null
-	result=$($processListCommand)
-	kill $(pgrep -l -f PhLTestImage.image |  cut -d ' ' -f1) >/dev/null
-    assertNotNull "$result"
+    launchSampleImageCommand> /dev/null
+    result=$(processListCommand)
+    kill $(pgrep -l -f $SAMPLEIMAGE.image |  cut -d ' ' -f1) >/dev/null
+    assertContains "$result" "$SAMPLEIMAGE"
+	
 }
 
 testLauncherKillAllCommandWithOneImageLaunchedShouldKillAll(){
-	$launchCommand >/dev/null
-	result=$($processListCommand)
-	assertNotNull "$result"
-	$killAllCommand
-	result=$($processListCommand)
-	assertNull "$result"
+	launchSampleImageCommand> /dev/null
+	result=$(processListCommand)
+	echo "actual: $result"
+    echo "expected (should contain): $SAMPLEIMAGE"
```

Comment:
Are the echos needed? I guess it is for debugging purposes when the assertion fails.
Does'nt the assertion provide the same kind of output? If not, it's too bad. If yes, probably echos could be removed.

Reply:
Well, when I tested, it shown only the expected value, but not actual. I can remove for now (since I used for debugging). Maybe to consider more verbose output. I didn't find in shunit library, if there are options to show actual.

Reply:
yes, it is std behaviour to ouput both expected and actual output when an assert fails. Maybe you can wrap the assert in a function to add missing echo, and reuse this function for asserting (less duplication)

Reply:
Maybe it was because the actual was empty and was shown on one line... I will try and if really missing, I can add wrapper function to show actual.